Library/Homebrew/os/linux/elf.rb

```diff
@@ -157,6 +145,49 @@ def needed_libraries_using_patchelf_rb(path)
       patcher = path.patchelf_patcher
       [patcher.soname, patcher.needed]
     end
+
+    def ld_so_conf(conf_path = "/etc/ld.so.conf")
```

Comment:
You will probably need to consider `HOMEBREW_PREFIX/etc/ld.so.conf` if Homebrew glibc is used

Reply:
> You will probably need to consider `HOMEBREW_PREFIX/etc/ld.so.conf` if Homebrew glibc is used

In the case that it is, should `/etc/ld.so.conf` still be considered (after considering `HOMEBREW_PREFIX/etc/ld.so.conf`) or should only `HOMEBREW_PREFIX/etc/ld.so.conf` be considered?

Reply:
It replaces.

`HOMEBREW_PREFIX/lib/ld.so` should point to the correct dynamic linker being used. You can run `HOMEBREW_PREFIX/lib/ld.so --list-diagnostics` and extract `path.sysconfdir` to get the `etc` directory.

Also relevant is `path.system_dirs`, which are used if nothing in the config files were found, _except_ if the executable sets `DF_1_NODEFLIB` in `DT_FLAGS_1` in which case not only should this step be skipped but anything in the config that have those directories as a prefix should also be skipped.

Reply:
> `HOMEBREW_PREFIX/lib/ld.so` should symlink to the correct dynamic linker being used whether that be system or Homebrew glibc. You can run `HOMEBREW_PREFIX/lib/ld.so --list-diagnostics` and extract `path.sysconfdir` to get the `etc` directory.

We now execute `HOMEBREW_PREFIX/lib/ld.so --list-diagnostics` to get `path.sysconfdir`. If `HOMEBREW_PREFIX/lib/ld.so` does not exist (for instance, on a fresh `brew` install before any other install command is executed - e.g. the initial state of the `brew` container), the fallback config directory `/etc` is used.

> Also relevant is path.system_dirs, which are used if nothing in the config files were found, except if the executable sets DF_1_NODEFLIB in DT_FLAGS_1 in which case not only should this step be skipped but anything in the config that have those directories as a prefix should also be skipped.

I _think_ I've designed this as intended, hopefully I understood the last part correctly ("anything in the config that have those directories as a prefix should also be skipped" >> "ignore any paths specified in `ld.so.conf.d` files that are (direct or indirect) children of any path listed in `path.system_dirs`".